From b4bf3c83c5ba5ded7ec2bb8ca7bd2fa7cd78eb97 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Mon, 20 Oct 2025 09:56:28 -0500
Subject: [PATCH 365/409] Revert "ntdll: HACK: Limit address space for Reverse:
 1999."

This reverts commit c1f2b81a98ed4116e7ab7bbc9fe8aa26c70549c9.
---
 dlls/ntdll/unix/virtual.c | 24 ------------------------
 1 file changed, 24 deletions(-)

diff --git a/dlls/ntdll/unix/virtual.c b/dlls/ntdll/unix/virtual.c
index ee44c4a1371..237ab8f117d 100644
--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -2495,28 +2495,6 @@ static void clear_native_views(void)
     }
 }
 
-static void fixup_effective_user_space_limit( const void **effective_user_space_limit )
-{
-#ifdef _WIN64
-    static int cached = -1;
-
-    if (cached == -1)
-    {
-        const char *e = getenv( "PROTON_LIMIT_ADDRESS_SPACE" );
-        const char *sgi = getenv( "SteamGameId" );
-        if (e) cached = e[0] != '0';
-        else cached = sgi &&
-            (
-                !strcmp( sgi, "3092660" )
-                || !strcmp( sgi, "3681810" )
-            );
-        if (cached) FIXME( "fixing up effective user space limit.\n" );
-    }
-    if (cached)
-        *effective_user_space_limit = min( *effective_user_space_limit, (void *)0x700000000000 );
-#endif
-}
-
 /***********************************************************************
  *           map_view
  *
@@ -2533,8 +2511,6 @@ static NTSTATUS map_view( struct file_view **view_ret, void *base, size_t size,
     const void *effective_user_space_limit = !is_win64 && wine_allocs_2g_limit ?
         (void *)0x7fff0000 : min( user_space_limit, host_addr_space_limit);
 
-    fixup_effective_user_space_limit( &effective_user_space_limit );
-
     if (alloc_type & MEM_REPLACE_PLACEHOLDER)
     {
         struct file_view *view;
-- 
2.51.1

